docs/content/config/best-practices/dkim_dmarc_spf.md

```diff
@@ -0,0 +1,347 @@
+
+    !!! bug inline end "DMS config volume support is not ready for Rspamd"
+
+        Presently you'll need to [explicitly mount `rspamd/modules/override.d/`][docs-rspamd-config-dropin] as an additional volume; do not use [`rspamd-modules.conf`][docs-rspamd-config-declarative] for this purpose.
+
+    ??? example "DKIM Signing Module Configuration Examples"
+
+        A simple configuration could look like this:
+
+        ```cf
+        # documentation: https://rspamd.com/doc/modules/dkim_signing.html
+
+        enabled = true;
+
+        sign_authenticated = true;
+        sign_local = true;
+
+        use_domain = "header";
+        use_redis = false; # don't change unless Redis also provides the DKIM keys
+        use_esld = true;
+        check_pubkey = true; # you wan't to use this in the beginning
+
+        domain {
+            example.com {
+                path = "/tmp/docker-mailserver/rspamd/dkim/mail.private";
+                selector = "mail";
+            }
+        }
+        ```
+
+        As shown next, you can:
+
+        - You can add more domains into the `domain { ... }` section.
+        - A domain can also be configured with multiple selectors and keys within a `selectors [ ... ]` array.
+
+        ```cf
+        # ...
+
+        domain {
+            example.com {
+                selectors [
+                    {
+                        path = "/tmp/docker-mailserver/rspamd/dkim/example.com/rsa.private";
+                        selector = "dkim-rsa";
+                    },
+                    {
+                        path = /tmp/docker-mailserver/rspamd/example.com/ed25519.private";
+                        selector = "dkim-ed25519";
+                    }
+                ]
+            }
+            example.org {
+                selectors [
+                    {
+                        path = "/tmp/docker-mailserver/rspamd/dkim/example.org/rsa.private";
+                        selector = "dkim-rsa";
+                    },
+                    {
+                        path = "/tmp/docker-mailserver/rspamd/dkim/example.org/ed25519.private";
+                        selector = "dkim-ed25519";
+                    }
+                ]
+            }
+        }
+        ```
+
+        !!! warning "Support for DKIM keys using Ed25519"
+
+            This modern elliptic curve is supported by Rspamd, but support by third-parties for [verifying Ed25519 DKIM signatures is unreliable][dkim-ed25519-support].
+
+            If you sign your mail with this key type, you should include RSA as a fallback, like shown in the above example.
+
+        !!! tip "DKIM Signing config: `check_pubkey = true;`"
+
+            This setting will have Rspamd query the DNS record for each DKIM selector, verifying each public key matches the private key configured.
+
+            If there is a mismatch, a warning will be omitted to the Rspamd log (`/var/log/supervisor/rspamd.log`).
```

Comment:
```suggestion
=== "OpenDKIM"

    OpenDKIM is currently [enabled by default][docs-env-opendkim].

    The command `docker exec <CONTAINER NAME> setup config dkim help` details supported config options, along with some examples.

    !!! example "Create a DKIM key"

        Generate the DKIM files with:

        ```sh
        docker exec -ti <CONTAINER NAME> setup config dkim
        ```

        Your new DKIM key(s) and OpenDKIM config files have been added to `/tmp/docker-mailserver/opendkim/`.

    ??? note "LDAP accounts need to specify domains explicitly"

        The command is unable to infer the domains from LDAP user accounts, you must specify them:

        ```sh
        setup config dkim domain 'example.com,example.io'
        ```

    ??? tip "Changing the key size"

        The private key presently defaults to RSA-4096. To create an RSA 2048-bit key run:

        ```sh
        setup config dkim keysize 2048
        ```

=== "Rspamd"

    Opt-in via [`ENABLE_RSPAMD=1`][docs-env-rspamd] (_and disable the default OpenDKIM: `ENABLE_OPENDKIM=0`_).

    Rspamd provides DKIM support through two separate modules:

    1. [Verifying DKIM signatures from inbound mail][rspamd-docs-dkim-checks] is enabled by default.
    2. [Signing outbound mail with your DKIM key][rspamd-docs-dkim-signing] needs additional setup (key + dns + config).

    !!! example "Create a DKIM key"

        Presently only OpenDKIM is supported with `setup config dkim`. To generate your DKIM key and DNS files you'll need to specify:

        - `-s` The DKIM selector (_eg: `mail`, it can be anything you like_)
        - `-d` The sender address domain (_everything after `@` from the email address_)

        See `rspamadm dkim_keygen -h` for an overview of the supported options.

        ---

        1. Go inside the container with `docker exec -ti <CONTAINER NAME> bash`
        2. Add `rspamd/dkim/` folder to your config volume and switch to it: `cd /tmp/docker-mailserver/rspamd/dkim`
        3. Run: `rspamadm dkim_keygen -s mail -b 2048 -d example.com -k mail.private > mail.txt` (_change `-d` to your domain-part_)
        4. Presently you must ensure Rspamd can read the `<selector>.private` file, run:
             -`chgrp _rspamd mail.private`
             -`chmod g+r mail.private`

    ---

    !!! bug inline end "DMS config volume support is not ready for Rspamd"

        Presently you'll need to [explicitly mount `rspamd/modules/override.d/`][docs-rspamd-config-dropin] as an additional volume; do not use [`rspamd-modules.conf`][docs-rspamd-config-declarative] for this purpose.

    Create a configuration file for the DKIM signing module at `rspamd/modules/override.d/dkim_signing.conf` and populate it with config as shown in the example below:

    ??? example "DKIM Signing Module Configuration Examples"

        A simple configuration could look like this:

        ```cf
        # documentation: https://rspamd.com/doc/modules/dkim_signing.html

        enabled = true;

        sign_authenticated = true;
        sign_local = true;

        use_domain = "header";
        use_redis = false; # don't change unless Redis also provides the DKIM keys
        use_esld = true;
        check_pubkey = true; # you wan't to use this in the beginning

        domain {
            example.com {
                path = "/tmp/docker-mailserver/rspamd/dkim/mail.private";
                selector = "mail";
            }
        }
        ```

        As shown next, you can:

        - You can add more domains into the `domain { ... }` section.
        - A domain can also be configured with multiple selectors and keys within a `selectors [ ... ]` array.

        ```cf
        # ...

        domain {
            example.com {
                selectors [
                    {
                        path = "/tmp/docker-mailserver/rspamd/dkim/example.com/rsa.private";
                        selector = "dkim-rsa";
                    },
                    {
                        path = /tmp/docker-mailserver/rspamd/example.com/ed25519.private";
                        selector = "dkim-ed25519";
                    }
                ]
            }
            example.org {
                selectors [
                    {
                        path = "/tmp/docker-mailserver/rspamd/dkim/example.org/rsa.private";
                        selector = "dkim-rsa";
                    },
                    {
                        path = "/tmp/docker-mailserver/rspamd/dkim/example.org/ed25519.private";
                        selector = "dkim-ed25519";
                    }
                ]
            }
        }
        ```

        !!! warning "Support for DKIM keys using Ed25519"

            This modern elliptic curve is supported by Rspamd, but support by third-parties for [verifying Ed25519 DKIM signatures is unreliable][dkim-ed25519-support].

            If you sign your mail with this key type, you should include RSA as a fallback, like shown in the above example.

        !!! tip "DKIM Signing config: `check_pubkey = true;`"

            This setting will have Rspamd query the DNS record for each DKIM selector, verifying each public key matches the private key configured.

            If there is a mismatch, a warning will be omitted to the Rspamd log (`/var/log/supervisor/rspamd.log`).
```